app/src/main/java/de/danoeh/antennapod/fragment/SearchFragment.java

```diff
@@ -241,14 +260,56 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
 
     @Override
     public boolean onContextItemSelected(@NonNull MenuItem item) {
+
+        int itemId = item.getItemId();
         FeedItem selectedItem = adapter.getLongPressedItem();
+        Feed selectedFeedItem  = adapterFeeds.getLongPressedItem();
+
+        if (selectedFeedItem != null) {
+            if (itemId == R.id.rename_folder_item) {
+                new RenameItemDialog(getActivity(), selectedFeedItem).show();
+                return true;
+            } else if (itemId == R.id.remove_all_inbox_item) {
+
+                DisplayConfirmationDialog.display(getActivity(), R.string.remove_all_inbox_label,
+                        R.string.remove_all_inbox_confirmation_msg,
+                        () -> DBWriter.removeFeedNewFlag(selectedFeedItem.getId()), new StatusListener() {
+                            @Override
+                            public void onActionSuccess() {}
+
+                            @Override
+                            public void onActionFailure(Throwable throwableError) {
+                                Log.e(TAG, Log.getStackTraceString(throwableError));
+                            }
+                        });
+                return true;
+            } else if (itemId == R.id.edit_tags) {
+                TagSettingsDialog.newInstance(Collections.singletonList(selectedFeedItem.getPreferences()))
+                        .show(getChildFragmentManager(), TagSettingsDialog.TAG);
+                return true;
+            } else if (itemId == R.id.rename_item) {
+                new RenameItemDialog(getActivity(), selectedFeedItem).show();
+                return true;
+            } else if (itemId == R.id.remove_feed) {
+                RemoveFeedDialog.show(getContext(), selectedFeedItem);
+                return true;
+            }
+            return super.onContextItemSelected(item);
```

Comment:
When long-pressing and then closing the dialog, the `selectedFeedItem` stays non-null. I would base this only on the `itemId`, not on whether `selectedFeedItem` is null.